lms/djangoapps/support/static/support/js/views/enrollment_modal.js

```diff
@@ -55,6 +55,18 @@
                         // Success callback
                         _.bind(function() {
                             this.hide();
+                            /*
+                            * `/certificates/regenerate` makes the certificate status to unavailable (if it exists)
```

Comment:
@adampalay `/certificates/regenerate` regenerates certificate if it exists otherwise generate the certificate. So this will still operate if no certificate was generated. 

So for audit, there would not be certificate generated, while for honor, it is my understanding then if user has passed the course and is eligible, only then he is eligible to get certificate. If honor user is eligible for certificate, this will generate the certificate for him. Is this the behaviour we want ?